Fixed issue #6

AYastrebov opened the pull request:
Fixed [issue #6](https://github.com/Ink/ThatInbox/issues/6) 
- Added NSString category for validating email address
- Automatically show keyboard, when compose view is shown

---

Kind regards,
AgileFusion

AYastrebov committed:
Fixed issue #6

AYastrebov committed:
Fixed issue #16

AYastrebov committed:
Fixed issue #5

AYastrebov committed:
Fixed #warning related to incorrect INK.framework integration with a project

AYastrebov committed:
updated mailcore2 library to latest version

AYastrebov committed:
removed appetitive and added mailcore2 as a git submodule

AYastrebov committed:
there was a bug in commit ca7251d1c19976cbcf8adfc9b27f5ddcf00d33fa

So i decided it would better to add mailcore2 and appetitive as a git submodules.
It should work now. To get submodules, you have to update your repository recursively

AYastrebov commented:
Please, be careful with updating. I've included mailcore2 and apptentive-ios as a git-submodules. 
I think it is a more better solution of using side iOS libraries which are rapidly developed (like mailcore2).

If you don't like this approach, feel free to let me know about it.

liyanchang commented:
Sounds fine. Been meaning to do it, but we should add a build script that grabs the submodules if you don't have it.

I'd like to keep this easy to use for first time developers and git submodules is not common knowledge. (not hard, just unnecessary friction)

AYastrebov committed:
Fixed issue #1

liyanchang commented on App/AppDelegate.m:
```diff
@@ -130,8 +133,31 @@ - (void) replyBlob:(INKBlob *)blob action:(INKAction*)action error:(NSError*)err
+    UISplitViewController *focused = (UISplitViewController *)[root focusedController];
+    UIViewController *presented = focused.presentedViewController;
+    
+    if (presented && [presented isKindOfClass:[AuthViewController class]])
```
Our convention has been:

```
if () {
   ...
}
```

Can you modify to match?

liyanchang commented on App/Helpers/AuthViewController.m:
```diff
@@ -0,0 +1,189 @@
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Presention
```
no big deal, but typo: Presentation

liyanchang commented:
Ideally, it would be something nicer then the default uialertview. We should be using the flat uialertview like we do with the logout dialog:

http://cl.ly/image/070R0z0t1H47
https://github.com/Grouper/FlatUIKit#alert-views

liyanchang commented:
Given that the auth screen (http://cl.ly/image/1b0b271c0p0C) is the first thing people see of that inbox, we'd like it to look good. So the header should look nicer (flat bar and arrows).

Here's a mockup.

Color for header bar: f1f1f1
Color for header text: 565656
Color for back button: 2980b9 (belizehole in uicolor+flatui.h)

No need for a forward button.
The back button shouldn't show up on the login page. Should only show up on other pages, like the can't access your account page.

![screen-shot-2013-09-17-at-3 30 18-pm](https://f.cloud.github.com/assets/328073/1161332/5fd2b786-1fea-11e3-86a1-7374c8eaae4c.png)

liyanchang commented:
When trying to scroll horizontal, you end up trigger the right hand menu. We should figure out how to drag on the message body while leaving the drag possible on the message list.

http://cl.ly/image/2D0C3z3d0c2D

Also, it should only scroll the message, not the headers

liyanchang commented:
It'll be easier if these are all separate pull requests.

liyanchang commented:
I know I had earlier commented that we should figure out how to make the git submodules autoload so it's easier. As it turns out, much harder then I expected. I tried a bunch of methods, including putting it as a preaction in the scheme, but I think it turned out that trying to get it to be magical was harder, confusing, and more error prone then just letting the user know about it. 

Anyways, we should add this run script to thatinbox (unless you can think of something slicker, which would be awesome!)
http://cl.ly/image/2R3l2W2X3m0M

```
# A quick test for if submodules exist. Not a perfect test, but should work for 99% of the cases
if [ ! -d "Vendor/mailcore2" ] || [ $(ls -A Vendor/mailcore2) ]
then
    echo 'error: git submodules not present.'
    echo 'To fix this: just run following commands in the terminal'
    echo ''
    echo '> cd '${PROJECT_DIR}
    echo '> git submodule init'
    echo '> git submodule update'
    exit 1
fi
```

liyanchang commented:
Thanks for all the help. Super excited to be able to push out these fixes. Let me know how I can help (sorry about the delay on the review; i'll be much faster next time, promise).

AYastrebov committed:
- updated mailcore2 submodule
- replaced UIAlertView with FlatAlertView
- Made navigationBar flat on authorization
- Removed forward button and moved back to the left

AYastrebov commented:
Please, take a look.
The run script doen't work for me. I have all submodules updated, but it fails with error.
We should figure out how to fix that.

liyanchang commented:
Ah. nvm. looks like you updated flatuikit. got it.

liyanchang commented:
Let's not worry about the build script in this pull request. already getting a bit too big. heh.

liyanchang commented:
Forgot that I can't push to this branch..... anyways, the changes are pretty straightforward.
1. Use flatuit hexcode

Not sure why your hex code color stuff is funky looking:
http://cl.ly/image/2v3G3X2d0s3H
Changed to use flatui which seems to fix it.

```

 #import "AuthNavigationViewController.h"
-#import "UIColor+FromHex.h"
+#import "UIColor+FlatUI.h"
 #import "UINavigationBar+FlatUI.h"
 #import <QuartzCore/CALayer.h>

@@ -73,7 +73,7 @@
         _activityIndicator.hidden = YES;

         // Configure FLATUI NavigationBar
-        [self.navigationBar configureFlatNavigationBarWithColor:[UIColor colorWithHexCode:@"f1f1f1"]];
+        [self.navigationBar configureFlatNavigationBarWithColor:[UIColor colorFromHexCode:@"F1F1F1"]];
```
1.  flat ui changed their button ordering.

```
index 6c42b5d..47085c6 100644
--- a/App/Menu/MenuViewController.m
+++ b/App/Menu/MenuViewController.m
@@ -235,7 +235,7 @@ bool firstLoad = YES;
 #pragma mark Logout Functions

 - (void)alertView:(FUIAlertView *)alertView clickedButtonAtIndex:(NSInteger)buttonIndex {
-    if (buttonIndex == 0){
+    if (buttonIndex == 1){
         [self logoutPressed];
     }
```
1. remove the fromhex category
2. remove the build script. it didn't work for me on a clean build either....

liyanchang commented:
One thing I didn't fix.
http://cl.ly/image/1L1C2p410V44

What's going on?

liyanchang commented:
Also, thanks for the work. Gave you push pull rights for the repo so once you finish these minor fixes, feel free to marge. When I write code for this, i'll ask you for a pull request and code review and vice versa.

AYastrebov commented:
Sorry for ugly alert, my mistake.

AYastrebov committed:
Fixed some issues from last commit

- fixed ugly AlertView
- Using FLatUI color from HEX
